packages/playwright-core/src/server/trace/recorder/tracing.ts

```diff
@@ -421,16 +421,16 @@ export class Tracing extends SdkObject implements InstrumentationListener, Snaps
     return { artifact };
   }
 
-  async _captureSnapshot(snapshotName: string, sdkObject: SdkObject, metadata: CallMetadata): Promise<void> {
+  _captureSnapshot(snapshotName: string, sdkObject: SdkObject, metadata: CallMetadata): Promise<void> | undefined {
     if (!this._snapshotter)
-      return;
+      return undefined;
     if (!sdkObject.attribution.page)
-      return;
+      return undefined;
     if (!this._snapshotter.started())
-      return;
+      return undefined;
     if (!shouldCaptureSnapshot(metadata))
-      return;
-    await this._snapshotter.captureSnapshot(sdkObject.attribution.page, metadata.id, snapshotName).catch(() => {});
+      return undefined;
+    return this._snapshotter.captureSnapshot(sdkObject.attribution.page, metadata.id, snapshotName).catch(() => {});
```

Comment:
This silent catch is triggered more often than you think. So you don't really make the downstream code drop any of the consistency checks, it can still stumble upon missing snapshots, and that is _fine_.